core/camel-util/src/main/java/org/apache/camel/util/ReflectionHelper.java

```diff
@@ -179,20 +179,10 @@ public static Method findMethod(Class<?> clazz, String name, Class<?>... paramTy
 
     public static void setField(Field f, Object instance, Object value) {
         try {
-            boolean oldAccessible = false;
-            try {
-                oldAccessible = f.canAccess(instance);
-            } catch (Exception e) {
-                // ignore
-            }
-            boolean shouldSetAccessible = !Modifier.isPublic(f.getModifiers()) && !oldAccessible;
-            if (shouldSetAccessible) {
+            if (!Modifier.isPublic(f.getModifiers()) && !f.canAccess(instance)) {
```

Comment:
Yes, that sounds better. Maybe, we could do the same with `getField`, don't you agree?